mllib/src/main/scala/org/apache/spark/ml/regression/GBTRegressor.scala

```diff
@@ -269,6 +269,21 @@ class GBTRegressionModel private[ml](
     new OldGBTModel(OldAlgo.Regression, _trees.map(_.toOld), _treeWeights)
   }
 
+  /**
+   * Method to compute error or loss for every iteration of gradient boosting.
+   *
+   * @param dataset Dataset for validation.
+   */
+  @Since("2.4.0")
+  def evaluateEachIteration(dataset: Dataset[_]): Array[Double] = {
```

Comment:
Do we want to support evaluation on other losses, as in the old API?  It might be nice to be able to without having to modify the Model's loss Param value.